src/db.ts

```diff
@@ -1,27 +1,204 @@
+        this.connect().catch((error) => {
+            console.error("Database connection error:", error);
+            process.exit(1);
+        });
+    }
+
+    public static getInstance(): dbHandler {
+        if (!dbHandler.instance) {
+            dbHandler.instance = new dbHandler();
+        }
+        return dbHandler.instance;
+    }
+
+    private async connect(): Promise<void> {
         await mongoose.connect(uri);
-        await mongoose.connection.db.admin().command({ ping: 1 });
-        console.log("successfully connected to DB");
-    } finally {
+    }
+
+    private async disconnect(): Promise<void> {
         await mongoose.disconnect();
     }
+
+    public async createRoom(roomData: RoomSchema): Promise<void> {
+        try {
+            const roomModel = getModelForClass(RoomSchema);
+            const room = new roomModel({ HostWsId: roomData.HostWsId, PlayerIds: roomData.PlayerIds, RoomState: roomData.RoomState});
+            await room.save();
+            console.log(`New room created with id (handler): ${room._id}`);
+        } catch (error) {
+            console.error("Room create error (handler):", error);
+        }
+    }
+
+    public async updateRoomRecord(roomId: string, updateData: {
+        RoomState?: string,
+        PlayerIds?: Array<string>,
+        HostWsId?: string
+    }): Promise<void> {
+        const roomModel = getModelForClass(RoomSchema);  
+        try {
+            const res = await roomModel.updateOne({ _id: roomId }, updateData);
+            console.log(`Room updated with id (handler): ${roomId}`);
+        } catch (error) {
+            console.error("Room update error (handler):", error);
+        }
+    }
+
+    public async createQuiz(quizData: QuizSchema): Promise<void> {
+        try {
+            const quizModel = getModelForClass(QuizSchema);
+            const quiz = new quizModel({ Questions: quizData.Questions });
+            await quiz.save();
+            console.log(`New quiz created with id (handler): ${quiz._id}`);
+        } catch (error) {
+            console.error("Quiz create error (handler):", error);
+        }
+    }
+
+    public async updateQuizList(quizId: string, updateData: {
+        Questions?: Array<string>,
+    }): Promise<void> {
+        const quizModel = getModelForClass(QuizSchema);
+        try {
+            const res = await quizModel.updateOne({ _id: quizId }, updateData);
+            console.log(`Quiz updated with id (handler): ${quizId}`);
+        } catch (error) {
+            console.error("Quiz update error (handler):", error);
+        }
+    }
+
+    public async createQuestionHandler(questionData: QuestionSchema): Promise<void> {
+        try {
+            const questionModel = getModelForClass(QuestionSchema);
+            const question = new questionModel({ Question: questionData.Question, PossibleAnswers: questionData.PossibleAnswers, CorrectAnswer: questionData.CorrectAnswer, QuestionType: questionData.QuestionType});
```

Comment:
[NIT] if `questionData` has the appropriate key-value pairs, then you can just use `questionData` as the argument when creating a question record